code/game/objects/structures/tables_racks.dm

```diff
@@ -227,71 +227,91 @@
 /obj/structure/table/item_interaction_secondary(mob/living/user, obj/item/tool, list/modifiers)
 	if(istype(tool, /obj/item/construction/rcd))
 		return NONE
-
 	if(istype(tool, /obj/item/toy/cards/deck))
-		var/obj/item/toy/cards/deck/dealer_deck = tool
-		if(HAS_TRAIT(dealer_deck, TRAIT_WIELDED)) // deal a card faceup on the table
-			var/obj/item/toy/singlecard/card = dealer_deck.draw(user)
-			if(card)
-				card.Flip()
-				attackby(card, user, list2params(modifiers))
-			return ITEM_INTERACT_SUCCESS
+		. = deck_act(user, tool, modifiers, TRUE)
+
+	// Continue to placing if we don't do anything else
+	if(.)
+		return .
```

Comment:
this feels ugly to me, just do

```dm
var/deck_act_value = NONE
if(istype(...))
 deck_act_value = NONE

if(deck_act_value == NONE)
 return NONE
```

really strange to flip flop on modifying `.` here imo. far cleaner this way too since we aren't treating a bitflag value like a boolean.